### ISSUE TITLE
22.7.9 breaks DHCPv4 for IPv6-enabled devices

### ISSUE THREAD

**Issue by no-usernames-left** [opnsense/core#6186]
22.7.9 is broken in some sort of horrible way -- this wasted HOURS of my time before I threw in the towel and invoked `opnsense-revert -r 22.7.7 opnsense` and rebooted, which instantly resolved the issue.

I updated on the evening of 5 December, and the network fell over on the evening of the 7th even though DHCP leases are two hours; I'm guessing the grace period is 48 hours, hence the delay in the failure.

Devices which were IPv6-capable were failing to get an IPv4 address via DHCP via Ethernet, regardless of whether IPv6 was enabled or disabled on the LAN and/or WAN interfaces, but still worked fine via Wi-Fi (and the AP dumps the traffic into the same VLAN, so what. the. fuck.)

Disabling IPv6 on end devices where possible (so, for Apple, forget it) instantly fixed the issue, but that was a workaround at best.

tl;dr: [bectl](https://geekcabi.net/posts/opnsense-boot-environments/) is your friend and I will absolutely be using that before every single update in the future, but this should be baked right into the OPNsense update flow.

**Comment by fichtner** [comment]
TBH, you will have to start making more sense and do less double-posting as that will not make it any more popular or technical... https://www.reddit.com/r/opnsense/comments/z9njkv/comment/izixq5y/?context=3

How are "IPv6-capable" devices interfering with IPv4 leases? It just sounds some form of DNS issue you haven't been willing to troubleshoot.


Cheers,
Franco

**Comment by ThePowerOfDreams** [comment]
@fichtner I held Cisco certification for years; I know the difference between DHCP and DNS.

The affected endpoints were sending DHCPDISCOVER and not receiving a DHCPOFFER. Reverting from 22.7.9 to 22.7.7 instantly solved the problem with zero config changes.

If that's not a smoking gun, I'm not sure what is.

>How are "IPv6-capable" devices interfering with IPv4 leases?

I have no idea, as my IPv6 skillz are not good. I'm just telling you what I found during my troubleshooting.

For example, QubesOS 4.1.1 wouldn't get an IPv4 address via DHCP until I disabled IPv6 on the ethernet interface -- and the instant I did, it was able to obtain a lease.

This is regardless of whether or not IPv6 was enabled on the internal VLAN, or the WAN interface, or both, or neither. None of the four possible permutations made a difference.

I crossposted for visibility in case others had the same issue, including the fix which worked for me.

**Comment by ThePowerOfDreams** [comment]
>TBH, you will have to start making more sense

In 25 years in IT I have never seen a failure mode like this. I thought I was dealing with hardware failure, but using a FreeBSD laptop with only IPv4 configured showed me that the hardware was fine at every step between the firewall and the affected endpoints. I was swapping PoE injectors, switches, you name it -- hours lost.

It was absolutely 100% resolved by simply downgrading to 22.7.7, which tells me where the problem was.

**Comment by fichtner** [comment]
Still being honest. We have these reports for EVERY update, however small, that something weird that cannot possibly happen happened.

What I can tell you is that things only happen for a very specific technical reason and this is where we can actually make bugfixes.

Today we know nothing other than allegedly 22.7.8 or 22.7.9 has an issue here. One that wasn't reported once before in the time that 22.7.8 and 22.7.9 have been out which is weeks worth of time for the same report to have surfaced by multiple people.

So for now I'm suspecting this is a local issue that needs more investment.


Cheers,
Franco

**Comment by ThePowerOfDreams** [comment]
I don't have time to collect packet traces; I will keep the unit on 22.7.7 until I must update, and I will create a BE before doing so.

Speaking of which, OPNsense should really create a BE before each update when running on ZFS, since reversion is often not possible without breakage. I got lucky this time as the delta was small.

**Comment by ThePowerOfDreams** [comment]
>Today we know nothing other than allegedly 22.7.8 or 22.7.9 has an issue here. One that wasn't reported once before in the time that 22.7.8 and 22.7.9 have been out which is weeks worth of time for the same report to have surfaced by multiple people.

[This comment on reddit](https://old.reddit.com/r/opnsense/comments/z9njkv/_/iyqwoy0/) is what made me start investigating IPv6 as a factor, and it was helpful even though it wasn't the scenario I was facing.

>Still being honest. We have these reports for EVERY update, however small, that something weird that cannot possibly happen happened.

I'm not disagreeing with you on this.

>What I can tell you is that things only happen for a very specific technical reason and this is where we can actually make bugfixes.

Also not disagreeing with this, but, there's genuinely something amiss here. If you want me to perform some tests, give me concrete instructions and I will do my best to get you data you can act on.
